### PR TITLE
Relax verify_first_message_correctness to accept any function call

### PR DESCRIPTION
**Pull request by vivi**
Useful for when you're running autogen agents (or running `--first`) and ask MemGPT to search archival memory right off the bat.

### CODE REVIEW

File: memgpt/agent.py [hunk 1]
```diff
@@ -466,18 +466,19 @@ def load_from_json_file_inplace(self, json_file):
             state = json.load(file)
         self.load_inplace(state)
 
-    def verify_first_message_correctness(self, response, require_send_message=True, require_monologue=False):
+    def verify_first_message_correctness(self, response, require_function_call=True, require_monologue=False):
```
Comment (vivi) [inline memgpt/agent.py:469]: Can change this argument name because `require_send_message` is not passed in any call to `verify_first_message_correctness`

**Comment by vivi** [comment]
```
❯ python3 agent_docs.py
LLM is explicitly disabled. Using MockLLM.
AgentConfig()
LLM is explicitly disabled. Using MockLLM.
LLM is explicitly disabled. Using MockLLM.
Generating embeddings: 0it [00:00, ?it/s]
new size 60
Saved local /Users/vivian/.memgpt/agents/agent_22/persistence_manager/index/nodes.pkl
Attached data source memgpt_paper to agent agent_22, consisting of 60. Agent now has 60 embeddings in archival memory.
LLM is explicitly disabled. Using MockLLM.
User_proxy (to chat_manager):

Tell me what a virtual context in MemGPT is. Search your archival memory.

--------------------------------------------------------------------------------
user :: {"type": "user_message", "message": {"content": "Tell me what a virtual context in MemGPT is. Search your archival memory.", "name": "User_proxy", "role": "user"}, "time": "2023-11-06 04:55:58 PM "}
No internal monologue but that's ok bc request_heartbeats
inner thoughts :: None
```

**Comment by cpacker** [comment]
Will this still fail on something like send_message with no inner monologue?

**Comment by vivi** [comment]
> Will this still fail on a `send_message` with no inner monologue?

Yes, it only passes if there is a function call and it chains (request_heartbeat=True). send_message doesn't have request_heartbeat iirc.

**Review by cpacker — APPROVED** [review]

**Pull request merged**